library/alloc/src/boxed/thin.rs

```diff
@@ -91,6 +93,83 @@ impl<T> ThinBox<T> {
 
 #[unstable(feature = "thin_box", issue = "92791")]
 impl<Dyn: ?Sized> ThinBox<Dyn> {
+    #[cfg(not(no_global_oom_handling))]
+    fn new_unsize_zst<T>(value: T) -> Self
+    where
+        T: Unsize<Dyn>,
+    {
+        #[repr(C)]
+        struct ReprC<A, B> {
+            a: A,
+            b: B,
+        }
+
+        // Allocate header like this:
+        // ```
+        // [ ... | header ]
```

Comment:
The `...` is padding. The thing here is that the padding is in front of the `H` instead of behind it, so that `ptr.sub(1)` from the pointer to the value returns the header.

Reply:
`H` is the first field, how can there be padding before it...?

Oh I realize there's two headers here, `AlignedHeader` and `H`. Maybe that's the confusion.

Reply:
Patched code comment here to make it more clear.